src/NServiceBus.Core/Transports/ErrorContext.cs

```diff
@@ -18,9 +18,17 @@ public class ErrorContext
         /// <param name="body">The message body.</param>
         /// <param name="transportTransaction">Transaction (along with connection if applicable) used to receive the message.</param>
         /// <param name="immediateProcessingFailures">Number of failed immediate processing attempts.</param>
-        /// <param name="context">A <see cref="ReadOnlyContextBag" /> which can be used to extend the current object.</param>
-        public ErrorContext(Exception exception, Dictionary<string, string> headers, string transportMessageId, byte[] body, TransportTransaction transportTransaction, int immediateProcessingFailures, ReadOnlyContextBag context)
+        /// <param name="context">A <see cref="ContextBag" /> which can be used to extend the current object.</param>
+        public ErrorContext(Exception exception, Dictionary<string, string> headers, string transportMessageId, byte[] body, TransportTransaction transportTransaction, int immediateProcessingFailures, ContextBag context)
         {
+            Guard.AgainstNull(nameof(exception), exception);
+            Guard.AgainstNull(nameof(headers), headers);
+            Guard.AgainstNullAndEmpty(nameof(transportMessageId), transportMessageId);
+            Guard.AgainstNull(nameof(body), body);
```

Comment:
I don't think we should be checking these here. All we are doing is passing them to `IncomingMessage`. Indeed, looking inside the constructor, we can see that these checks here are redundant.
```suggestion
```